addons/binding/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/handler/TeslaHandler.java

```diff
@@ -694,21 +711,21 @@ public void run() {
 
     protected Runnable eventRunnable = new Runnable() {
 
-        protected boolean establishEventStream() throws Exception {
+        protected void establishEventStream() throws Exception {
             eventClient = ClientBuilder.newClient()
                     .register(new Authenticator((String) getConfig().get(USERNAME), vehicle.tokens[0]));
             eventTarget = eventClient.target(TESLA_EVENT_URI).path(vehicle.vehicle_id + "/").queryParam("values",
                     StringUtils.join(EventKeys.values(), ',', 1, EventKeys.values().length));
             eventResponse = eventTarget.request(MediaType.TEXT_PLAIN_TYPE).get();
+
+            logger.trace("Establishing the event stream : Response : {}", eventResponse.getStatusInfo());
+
             if (eventResponse.getStatus() == 200) {
                 InputStream dummy = (InputStream) eventResponse.getEntity();
-                // eventInputStreamReader = new
-                // InputStreamReader(eventResponse.readEntity(InputStream.class));
                 eventInputStreamReader = new InputStreamReader(dummy);
                 eventBufferedReader = new BufferedReader(eventInputStreamReader);
-                return true;
             } else {
-                return false;
+                throw new Exception();
```

Comment:
My Tesla branch was still in an old IDE setup (e.g. pre-OH2/ESH setup you provided with the Beta release). Time to ditch that and start with a clean slate